src/main/java/stsjorbsmod/actions/ExhaustNegativeCardAction.java

```diff
@@ -0,0 +1,42 @@
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+
+import java.util.ArrayList;
+
+// Based on ExhaustSpecificCardAction
+public class ExhaustNegativeCardAction extends AbstractGameAction {
+    private CardGroup pile;
+
+    public ExhaustNegativeCardAction(AbstractCreature source, CardGroup pile) {
+        this.setValues(AbstractDungeon.player, source, 1);
+        this.actionType = ActionType.EXHAUST;
+        this.duration = Settings.ACTION_DUR_FASTER;
+        this.pile = pile;
+    }
+
+    @Override
+    public void update() {
+        ArrayList<AbstractCard> toExhaust = new ArrayList<AbstractCard>();
```

Comment:
Generally, an `Action`'s update method gets called once per game-update-tick until either its duration has expired (via `tickDuration()` calls that each decrement it by game tick durations) or its update method sets `isDone = true`. Usually, actions will either be:

* "instant": they don't set `this.duration` in their ctor, and their update method just does its thing and unconditionally sets `isDone=true` - usually you see this for actions that work by queuing other action(s) that are non-instant, or
* "non-instant" (they set `this.duration = this.startingDuration = DUR_WHATEVER` in their ctor, they call `this.tickDuration()` unconditionally at the end of their update method, and the actual work is hidden behind a `if (this.duration == this.startingDuration) { /* do work */ }` clause so it only happens once.

This is working like a non-instant action but isn't following the usual pattern; I'd expect it to work fine anyway because its work happens to be idempotent, but I think it would be good to follow the usual pattern anyway (like `ExhaustSpecificCardAction` does) for consistency.

Reply:
Ewww, duration is a float. I can follow the base game's pattern, which should work because those variables are initialized to be equal, but that bit of it is ugly.

Reply:
Do you have any preference for use of "this."? My slow drift is to not use it except in constructors.

Reply:
I don't really have a preference on "this", since IntelliJ does a reasonable job of coloring locals/fields differently and warning on unused assignments.